runtimevar/filevar/filevar.go

```diff
@@ -88,34 +89,31 @@ func NewVariable(file string, decoder *runtimevar.Decoder, opts *WatchOptions) (
 
 // watcher implements driver.Watcher for configurations stored in files.
 type watcher struct {
-	notifier   *fsnotify.Watcher
-	file       string
-	decoder    *runtimevar.Decoder
-	bytes      []byte
-	isDeleted  bool
-	waitTime   time.Duration
-	updateTime time.Time
+	notifier *fsnotify.Watcher
+	file     string
+	decoder  *runtimevar.Decoder
+	waitTime time.Duration
 }
 
-// WatchVariable blocks until the file changes, the Context's Done channel closes or an error occurs.  It
-// will return an error if the configuration file is deleted, however, if it has previously returned
-// an error due to missing configuration file, the next WatchVariable call will block until the file has
-// been recreated.
-func (w *watcher) WatchVariable(ctx context.Context) (driver.Variable, error) {
-	zeroVar := driver.Variable{}
-	// Check for Context cancellation first before proceeding.
-	select {
-	case <-ctx.Done():
-		return zeroVar, ctx.Err()
-	default:
-		// Continue.
+func (w *watcher) WatchVariable(ctx context.Context, prevVersion interface{}, prevErr error) (*driver.Variable, interface{}, time.Duration, error) {
+
+	// checkSameErr checks to see if err is the same as prevErr, and if so, returns
+	// the "no change" signal with w.waitTime.
```

Comment:
Let's add a TODO here to revisit when we make the interface change we discussed. Comparing errors in this way is brittle.